android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java

```diff
@@ -87,7 +90,10 @@ private void importFile(Uri fileUri) {
 					outputStream.close();
 			}
 		} catch (IOException exception) {
-			Log.e("importFile", Objects.requireNonNull(exception.getMessage()));
+			String message = exception.getMessage();
+			if (message == null)
+				return;
+			Log.e("importFile", message);

```

Comment:
When message is null, this call shouldn´t break but maybe would be more informative to print the entire exception so will output the stacktrace.

```suggestion
if (message == null) {
	Log.e("importFile", exception);
} else {
	Log.e("importFile", message);
}
```
Or simpler, we can ignore the message and print the exception only.
```java
	Log.e("importFile", exception);
```

Reply:
Thanks, this is my first time writing Java code :D

Reply:
Apparently that isn't possible:

```
/home/runner/work/devilutionX/devilutionX/android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:97: error: incompatible types: IOException cannot be converted to String

	Log.e("importFile", exception);
> Task :app:compileDebugJavaWithJavac FAILED
```

Reply:
try `log.e("importFile", "IOException", exception);`? https://developer.android.com/reference/android/util/Log#e(java.lang.String,%20java.lang.String,%20java.lang.Throwable)

Reply:
@AJenbo sorry, @ephphatha is right, Log.e requires a third param to print an exception.
```java
if (message == null) {
	Log.e("importFile", "IOException", exception);
} else {
	Log.e("importFile", message);
}
```